lib/screens/main_screen.dart

```diff
@@ -64,9 +68,103 @@ class _MainScreenState extends State<MainScreen> {
   List<Token> _tokenList = List<Token>();
   final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
 
-  _MainScreenState() {
-    _loadAllTokens();
-    _loadFirebase();
+  Timer _pollTimer;
+
+  @override
+  void initState() {
+    super.initState();
+
+    // Start polling timer
+    SchedulerBinding.instance.addPostFrameCallback((_) async {
+      AppSettings.of(context).streamEnablePolling().listen(
+        (bool event) {
+          if (event) {
+            log('Polling is enabled.', name: 'main_screen.dart');
+            _pollTimer = Timer.periodic(
+                Duration(seconds: 10), (_) => _pollForRequests());
+          } else {
+            log('Polling is disabled.', name: 'main_screen.dart');
+            _pollTimer?.cancel();
+          }
+        },
+        cancelOnError: false,
+        onError: (error) => print('$error'),
+      );
+    });
+
+    // Load UI elements
+    SchedulerBinding.instance.addPostFrameCallback((_) => _loadEverything());
+  }
+
+  _pollForRequests() async {
+    // Get all push tokens
+    List<PushToken> pushTokens = (await StorageUtil.loadAllTokens())
+        .whereType<PushToken>()
+        .where((t) =>
+            t.isRolledOut &&
+            t.url !=
+                null) // Legacy tokens can not pull, because the url is missing!
+        .toList();
+
+    // Disable polling if no push tokens exist
+    if (pushTokens.isEmpty) {
+      log('No push token is available for polling, polling is disabled.',
+          name: 'main_screen.dart');
+      AppSettings.of(context).setEnablePolling(false);
+      return;
+    }
+
+    // Start request for each token
+    for (PushToken p in pushTokens) {
+      String timestamp = DateTime.now().toUtc().toIso8601String();
+
+      // Legacy android tokens are signed differently
+      String message = '${p.serial}|$timestamp';
+      String signature = p.privateTokenKey == null
+          ? await Legacy.sign(p.serial, message)
+          : createBase32Signature(p.getPrivateTokenKey(), utf8.encode(message));
+
+      Map<String, String> parameters = {
+        'serial': p.serial,
+        'timestamp': timestamp,
+        'signature': signature,
+      };
+
+      try {
+        Response response = await doGet(
+            url: p.url, parameters: parameters, sslVerify: p.sslVerify);
+
+        if (response.statusCode == 200) {
+          // The signature of this message must not be verified as each push
+          // request gets verified independently.
+          Map<String, dynamic> result = jsonDecode(response.body)['result'];
+          List values = result['value'];
+
+          for (Map<String, dynamic> value in values) {
+            _handleIncomingAuthRequest({'data': value});
+          }
+        } else {
+          // Who knows what happened here?
+          // TODO Handle this case?
```

Comment:
```suggestion
          // Error messages can only be distinguished by their text content, not by their error code. This would make error handling complex.
```